src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -141,7 +141,7 @@ public static class DependencyGraphRestoreUtility
                 cacheContextModifier(sourceCacheContext);
 
                 // Settings passed here will be used to populate the restore requests.
-                var restoreContext = GetRestoreContext(context, providerCache, sourceCacheContext, sources, dgFile, false); // We don't force in preview 
+                var restoreContext = GetRestoreContext(context, providerCache, sourceCacheContext, sources, dgFile, forceRestore : true);
```

Comment:
Preview is used only on install. So the no-op scenario should've never been hit, except for the above corner case.
No perf concerns here.